### PR TITLE
Retaining metadata type value given by user for device

### LINKED ISSUE #711: Device Type Metadata getting overwritten to "string" type
**What happened**:
Device instance Metadata gets overwritten to "string" value even if it is created with "int" value

**What you expected to happen**:
The metadata type given by user should be retained instead of overwriting it

**How to reproduce it (as minimally and precisely as possible)**:
1. Setup kubeedge cloud and edge
2. Create device model and a device instance with desired metadata type as int
3. Check the device value at edge using MQTT (it would have converted the metadata value to string)

**Environment**:
- KubeEdge version: 0.3

### PR DESCRIPTION
**Pull request by sujithsimon22**
**What type of PR is this?**
 /kind bug

**What this PR does / why we need it**:
This PR retains the value of metadata type if given by user else assigns it as string

**Which issue(s) this PR fixes**:

Fixes #711 #681

### CODE REVIEW

**Comment by sujithsimon22** [comment]
cc @rohitsardesai83 @sids-b for review

**Comment by subpathdev** [comment]
This will also fix #681

File: cloud/pkg/devicecontroller/controller/downstream.go [hunk 1]
```diff
@@ -400,6 +400,10 @@ func createDevice(device *v1alpha1.Device) types.Device {
 	for i, dtwin := range device.Status.Twins {
 		expected := &types.TwinValue{}
 		expected.Value = &device.Status.Twins[i].Desired.Value
+		metadataType, ok := device.Status.Twins[i].Desired.Metadata["type"]
```
Comment (sids-b) [inline cloud/pkg/devicecontroller/controller/downstream.go]: Does CRD definition support "Type" as well ??

Comment (subpathdev) [reply cloud/pkg/devicecontroller/controller/downstream.go]: no we are using a map there.
https://github.com/kubeedge/kubeedge/blob/master/cloud/pkg/devicecontroller/apis/devices/v1alpha1/device_instance_types.go#L145

Comment (sujithsimon22) [reply cloud/pkg/devicecontroller/controller/downstream.go]: No the CRD definition is generic and accepts map[string]string 
But we given an examples in sample crds which mentions the same https://github.com/kubeedge/kubeedge/blob/1a0d4e230e23036fc584991a95f06c968422e268/build/crd-samples/devices/create-device-instance.yaml#L34

Comment (sids-b) [reply cloud/pkg/devicecontroller/controller/downstream.go]: OK

Comment (sids-b) [reply cloud/pkg/devicecontroller/controller/downstream.go]: We need to mention all these points in usage docs. It can be done in future PRs

Comment (sujithsimon22) [reply cloud/pkg/devicecontroller/controller/downstream.go]: +1

**Pull request merged**